circuit-conversations-list.js

```diff
@@ -0,0 +1,237 @@
+const template = document.createElement('template');
+template.innerHTML = `
+<style>
+#conversationList {
+  overflow-y: auto;
+  min-width: var(--min-width);
+  max-height: var(--max-height);
+}
+.circuit-conversation {
+  display: flex;
+  align-items: center;
+  font-family: var(--conversation-font);
+  font-size: var(--conversation-font-size);
+  height: var(--conversation-height);
+  border: var(--conversation-border);
+  padding: var(--conversation-padding);
+}
+</style>
+<select class="select-wide" 
+        id="conversationList">
+</select>`;
+export class CircuitConversationsList extends HTMLElement {
+
+  set client(client) {
+    if (!client || (this._client && this._client.loggedOnUser && client.loggedOnUser.userId === this._client.loggedOnUser.userId)) {
+      return;
+    }
+    this._client = client;
+  }
+
+  set _connected(isConnected) {
+    if (isConnected) {
+      this.setAttribute('connected', '');
+    } else {
+      this.removeAttribute('connected');
+    }
+  }
+
+  get connected() {
+    return this.hasAttribute('connected');
+  }
+
+  get client() {
+    return this._client;
+  }
+
+  constructor() {
+    super();
+    this._client = null;
+
+    this.root = this.attachShadow({ mode: 'open' });
+    this.root.appendChild(template.content.cloneNode(true));
+
+    this._usersHashMap = {}; // Hashmap to store users names for the conversation feed
+    this._conversationsListElement = this.root.getElementById('conversationList'); // Conversation List Container
+    this._conversationsListElement.size = !!this.getAttribute('size') && Number(this.getAttribute('size')) || 5;
+  }
+
+  // Delay Circuit initialization to speed up page load. This way the SDK
+  // can be loaded with 'async'
+  async _init() {
+    Circuit.logger.setLevel(1);
+    this._client = Circuit.Client({
+      domain: this._domain,
+      client_id: this._clientId
+    });
+
+    // Event listeners
+
+    this._conversationsListElement.addEventListener('change', evt => {
+      const convId = evt.target.value;
+      const conversation = this._conversationsList.find(c => c.convId === convId);
+      this.dispatchEvent(new CustomEvent('selected', { 
+        detail: {
+          client: this._client,
+          conversation: conversation
+        }
+      }));
+    });
+
+    this._client.addEventListener('conversationUpdated', evt => {
+      const conversation = evt.conversation;
+      const index = this._conversationsList.findIndex(c => c.convId === conversation.convId);
+      if (index > -1 && conversation.type !== Circuit.Enums.ConversationType.DIRECT) {
+        this._conversationsList[index] = conversation;
+        const element = this.root.getElementById(conversation.convId);
+        element.innerHTML = conversation.topic || conversation.topicPlaceholder;
+        console.log('Conversation updated', conversation);
```

Comment:
We shouldn't have console.logs. Just console.errors.
I know there are probably a couple in the call components as well, please remove those too.